From patchwork Fri Nov 29 16:32:32 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11267341
Return-Path: <SRS0=D48H=ZV=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 7E12D14B7
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 29 Nov 2019 16:32:48 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 5D6712176D
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 29 Nov 2019 16:32:48 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="aiFTFfB9"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727110AbfK2Qcn (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 29 Nov 2019 11:32:43 -0500
Received: from us-smtp-delivery-1.mimecast.com ([205.139.110.120]:55236 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1727076AbfK2Qcn (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 29 Nov 2019 11:32:43 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575045162;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=C/xSejyaXzbtUL047a9u4nhEXzM8MndtKpPyPJyLXyY=;
        b=aiFTFfB93IDnCDUkPEswixX10FEAVKLviZKhBLPfg5MP0XOAQEPeTYJOSgNVe1mOnXraaX
        F+YWXahMyo0X6DfGgcEO3CeNH5A+RXYHFRtR3nvI4VuX1IDFo0owBJLoEwOzdJ1kEctJFl
        ngnOD42wrhoQZtB3eXSTSy2oeqIkclQ=
Received: from mail-qt1-f199.google.com (mail-qt1-f199.google.com
 [209.85.160.199]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-28-IecRPUdPMdW86Qnmx3jKLg-1; Fri, 29 Nov 2019 11:32:38 -0500
Received: by mail-qt1-f199.google.com with SMTP id j18so19222853qtp.15
        for <kvm@vger.kernel.org>; Fri, 29 Nov 2019 08:32:38 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=JfYuhM68fjTSb2nWUh/a1XhHzPawVsK+0KYnkYwzzyg=;
        b=oum5uABSqb+gkdxI9KvLDm9x+0QBCnPscQ+zhc78v5QnLL1NgsCAN8nxPf5bRZLylT
         GPGXmvEZGBWCoJ8/ao6TyvTXblB1UzeMgPWo9YG6iVCejmTNR54tWFtiBw4cAxQiFVyx
         t3KOA/OGQHdvAZzqFEaCdgSevxTe9odlreyKv9tjhsGrAn/48jOxd58IVdzDue6usEmU
         QXB92sH/ZaQQATAiSrDKmETMbnTra99OcGYvV2NqoFhnDheS1mQU6Jc4WeWW+qr7Xv3U
         TKJ+Nb2YDu8E8kczfl1qSHsXaXrpd5FsHROdDqhcSZT5s8OetBzUXH5Ols+npPnkc/4y
         k7Lw==
X-Gm-Message-State: APjAAAUKbTU9fz8t/1YWHlcFGsNpl7BVw0BYXr6vefrtxZiSLko+pOp3
        VmaeBhoiNaB2ZcxPYba7aZet2MAdGVL1pUXJka+n5hj1XosJPKrMigqlW6JdXqQJ39IwipX1mlH
        nBKBtDHMcZwFL
X-Received: by 2002:a0c:ee91:: with SMTP id
 u17mr16535786qvr.245.1575045158014;
        Fri, 29 Nov 2019 08:32:38 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqzmI9LDbhfjPk/kPBygYFre0p1EfkrO+8gFy85TuTM9UlPya9CAD4HYy130ni/hYFTuHHtkNw==
X-Received: by 2002:a0c:ee91:: with SMTP id
 u17mr16535759qvr.245.1575045157801;
        Fri, 29 Nov 2019 08:32:37 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 d9sm4568329qtj.52.2019.11.29.08.32.36
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 29 Nov 2019 08:32:36 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v2 1/3] KVM: X86: Some cleanups in ioapic.h/lapic.h
Date: Fri, 29 Nov 2019 11:32:32 -0500
Message-Id: <20191129163234.18902-2-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191129163234.18902-1-peterx@redhat.com>
References: <20191129163234.18902-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: IecRPUdPMdW86Qnmx3jKLg-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Both kvm_apic_match_dest() and kvm_irq_delivery_to_apic() should
probably suite more to lapic.h comparing to ioapic.h, moving.

kvm_apic_match_dest() is defined twice, once in each of the header.
Removing the one defined in ioapic.h.

Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/ioapic.h | 6 ------
 arch/x86/kvm/lapic.h  | 5 ++++-
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/arch/x86/kvm/ioapic.h b/arch/x86/kvm/ioapic.h
index ea1a4e0297da..2fb2e3c80724 100644
--- a/arch/x86/kvm/ioapic.h
+++ b/arch/x86/kvm/ioapic.h
@@ -116,9 +116,6 @@ static inline int ioapic_in_kernel(struct kvm *kvm)
 }
 
 void kvm_rtc_eoi_tracking_restore_one(struct kvm_vcpu *vcpu);
-bool kvm_apic_match_dest(struct kvm_vcpu *vcpu, struct kvm_lapic *source,
-		int short_hand, unsigned int dest, int dest_mode);
-int kvm_apic_compare_prio(struct kvm_vcpu *vcpu1, struct kvm_vcpu *vcpu2);
 void kvm_ioapic_update_eoi(struct kvm_vcpu *vcpu, int vector,
 			int trigger_mode);
 int kvm_ioapic_init(struct kvm *kvm);
@@ -126,9 +123,6 @@ void kvm_ioapic_destroy(struct kvm *kvm);
 int kvm_ioapic_set_irq(struct kvm_ioapic *ioapic, int irq, int irq_source_id,
 		       int level, bool line_status);
 void kvm_ioapic_clear_all(struct kvm_ioapic *ioapic, int irq_source_id);
-int kvm_irq_delivery_to_apic(struct kvm *kvm, struct kvm_lapic *src,
-			     struct kvm_lapic_irq *irq,
-			     struct dest_map *dest_map);
 void kvm_get_ioapic(struct kvm *kvm, struct kvm_ioapic_state *state);
 void kvm_set_ioapic(struct kvm *kvm, struct kvm_ioapic_state *state);
 void kvm_ioapic_scan_entry(struct kvm_vcpu *vcpu,
diff --git a/arch/x86/kvm/lapic.h b/arch/x86/kvm/lapic.h
index 39925afdfcdc..19b36196e2ff 100644
--- a/arch/x86/kvm/lapic.h
+++ b/arch/x86/kvm/lapic.h
@@ -83,7 +83,10 @@ int kvm_lapic_reg_read(struct kvm_lapic *apic, u32 offset, int len,
 		       void *data);
 bool kvm_apic_match_dest(struct kvm_vcpu *vcpu, struct kvm_lapic *source,
 			   int short_hand, unsigned int dest, int dest_mode);
-
+int kvm_apic_compare_prio(struct kvm_vcpu *vcpu1, struct kvm_vcpu *vcpu2);
+int kvm_irq_delivery_to_apic(struct kvm *kvm, struct kvm_lapic *src,
+			     struct kvm_lapic_irq *irq,
+			     struct dest_map *dest_map);
 bool __kvm_apic_update_irr(u32 *pir, void *regs, int *max_irr);
 bool kvm_apic_update_irr(struct kvm_vcpu *vcpu, u32 *pir, int *max_irr);
 void kvm_apic_update_ppr(struct kvm_vcpu *vcpu);

From patchwork Fri Nov 29 16:32:33 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11267345
Return-Path: <SRS0=D48H=ZV=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id C5D4D930
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 29 Nov 2019 16:32:58 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 9ACAD217AB
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 29 Nov 2019 16:32:58 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="ETwu8JJW"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727117AbfK2Qc5 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 29 Nov 2019 11:32:57 -0500
Received: from us-smtp-2.mimecast.com ([205.139.110.61]:21932 "EHLO
        us-smtp-delivery-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL)
        by vger.kernel.org with ESMTP id S1727070AbfK2Qcn (ORCPT
        <rfc822;kvm@vger.kernel.org>); Fri, 29 Nov 2019 11:32:43 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575045161;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=Hbyl2iZZRhb5QuVojk7TiwphUcR02e+q5G1tP0VBGU4=;
        b=ETwu8JJWbxjlFB9Y9QbLmbGaiwfhLOR+iDuodR9zccMMLHnaRzcJS3VgQIW0SCynweNZj2
        eL97sPDu9nND5czw6ySBGgNUmFWArvvp4st6MXi2zOAVfpfK8STVNK3Sy2CYKBouuJ3F/p
        wDAaQMiyMFK/TpIqekOLCU/4m1tGJRs=
Received: from mail-qv1-f72.google.com (mail-qv1-f72.google.com
 [209.85.219.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-104-CuU_0zXyNVqmwWcNJVU70w-1; Fri, 29 Nov 2019 11:32:40 -0500
Received: by mail-qv1-f72.google.com with SMTP id a4so13060438qvn.14
        for <kvm@vger.kernel.org>; Fri, 29 Nov 2019 08:32:40 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=0pt/Az8Wvxgx7uazfZToKoKUqCiGtbwmyJfL5cUmcEs=;
        b=Sp5wimZ3ZvdmiMw47UQEMo8puP6eEo3hp0w7NzYD7Ka1Uer424XpAplSBF+WN++aVd
         gtrnN3MJ9jJ4WPjQk30pQSAnpCCuUTQnlWp3rFCIp4GhACpfWWHiBNBkE/ZtNm1LEL/Z
         3lKcyBSBuhyHKUSg16ftOOv7gpViSWT+vZGmyV1OCpEM+LAINj8/0F8y1pXVQjPrtecZ
         RTdkQOaX6L02Zb/r9Diw5Rru3Ux7lcUJt0DdK09y2kx4NhEMsoN8OR3vkDCh9Y0YuFWK
         UcCusCsIChyN07XGU42PSO1SkqNfZOTbqQd8ES0dEdFE87MxxHdFmfl+oBw1l6bCXgLC
         on0g==
X-Gm-Message-State: APjAAAWKd60URTcKzB7bqkrSPt/0yDWg8l2PVCCppeCejFSWCyVvJDoo
        laKQjyRyhmEHeHn/ix9Jm8Hyqb2uecews3FlxDqr5Jk7X/j0oj29iVKzC7XlbrRa8Qh/y9McmTY
        2obw6XU7JIg58
X-Received: by 2002:a37:8a06:: with SMTP id m6mr17328074qkd.86.1575045159732;
        Fri, 29 Nov 2019 08:32:39 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqyIatg4wHEW2JAb8sx6aQy+GlrVkTSUrwO3DQB72FuUrIsdYmHKwuBkFTMbo+c/NlFAlPM5nQ==
X-Received: by 2002:a37:8a06:: with SMTP id m6mr17328032qkd.86.1575045159421;
        Fri, 29 Nov 2019 08:32:39 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 d9sm4568329qtj.52.2019.11.29.08.32.37
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 29 Nov 2019 08:32:38 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v2 2/3] KVM: X86: Use APIC_DEST_* macros properly
Date: Fri, 29 Nov 2019 11:32:33 -0500
Message-Id: <20191129163234.18902-3-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191129163234.18902-1-peterx@redhat.com>
References: <20191129163234.18902-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: CuU_0zXyNVqmwWcNJVU70w-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Previously we were using either APIC_DEST_PHYSICAL|APIC_DEST_LOGICAL
or 0|1 to fill in kvm_lapic_irq.dest_mode, and it's done in an adhoc
way.  It's fine imho only because in most cases when we check against
dest_mode it's against APIC_DEST_PHYSICAL (which equals to 0).
However, that's not consistent, majorly because APIC_DEST_LOGICAL does
not equals to 1, so if one day we check irq.dest_mode against
APIC_DEST_LOGICAL we'll probably always get a false returned.

This patch replaces the 0/1 settings of irq.dest_mode with the macros
to make them consistent.

Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/ioapic.c   | 9 ++++++---
 arch/x86/kvm/irq_comm.c | 7 ++++---
 arch/x86/kvm/x86.c      | 2 +-
 3 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/arch/x86/kvm/ioapic.c b/arch/x86/kvm/ioapic.c
index 9fd2dd89a1c5..1e091637d5d5 100644
--- a/arch/x86/kvm/ioapic.c
+++ b/arch/x86/kvm/ioapic.c
@@ -331,7 +331,8 @@ static void ioapic_write_indirect(struct kvm_ioapic *ioapic, u32 val)
 			irq.vector = e->fields.vector;
 			irq.delivery_mode = e->fields.delivery_mode << 8;
 			irq.dest_id = e->fields.dest_id;
-			irq.dest_mode = e->fields.dest_mode;
+			irq.dest_mode = e->fields.dest_mode ?
+			    APIC_DEST_LOGICAL : APIC_DEST_PHYSICAL;
 			bitmap_zero(&vcpu_bitmap, 16);
 			kvm_bitmap_or_dest_vcpus(ioapic->kvm, &irq,
 						 &vcpu_bitmap);
@@ -343,7 +344,8 @@ static void ioapic_write_indirect(struct kvm_ioapic *ioapic, u32 val)
 				 * keep ioapic_handled_vectors synchronized.
 				 */
 				irq.dest_id = old_dest_id;
-				irq.dest_mode = old_dest_mode;
+				irq.dest_mode = old_dest_mode ?
+				    APIC_DEST_LOGICAL : APIC_DEST_PHYSICAL;
 				kvm_bitmap_or_dest_vcpus(ioapic->kvm, &irq,
 							 &vcpu_bitmap);
 			}
@@ -369,7 +371,8 @@ static int ioapic_service(struct kvm_ioapic *ioapic, int irq, bool line_status)
 
 	irqe.dest_id = entry->fields.dest_id;
 	irqe.vector = entry->fields.vector;
-	irqe.dest_mode = entry->fields.dest_mode;
+	irqe.dest_mode = entry->fields.dest_mode ?
+	    APIC_DEST_LOGICAL : APIC_DEST_PHYSICAL;
 	irqe.trig_mode = entry->fields.trig_mode;
 	irqe.delivery_mode = entry->fields.delivery_mode << 8;
 	irqe.level = 1;
diff --git a/arch/x86/kvm/irq_comm.c b/arch/x86/kvm/irq_comm.c
index 8ecd48d31800..673b6afd6dbf 100644
--- a/arch/x86/kvm/irq_comm.c
+++ b/arch/x86/kvm/irq_comm.c
@@ -52,8 +52,8 @@ int kvm_irq_delivery_to_apic(struct kvm *kvm, struct kvm_lapic *src,
 	unsigned long dest_vcpu_bitmap[BITS_TO_LONGS(KVM_MAX_VCPUS)];
 	unsigned int dest_vcpus = 0;
 
-	if (irq->dest_mode == 0 && irq->dest_id == 0xff &&
-			kvm_lowest_prio_delivery(irq)) {
+	if (irq->dest_mode == APIC_DEST_PHYSICAL &&
+	    irq->dest_id == 0xff && kvm_lowest_prio_delivery(irq)) {
 		printk(KERN_INFO "kvm: apic: phys broadcast and lowest prio\n");
 		irq->delivery_mode = APIC_DM_FIXED;
 	}
@@ -114,7 +114,8 @@ void kvm_set_msi_irq(struct kvm *kvm, struct kvm_kernel_irq_routing_entry *e,
 		irq->dest_id |= MSI_ADDR_EXT_DEST_ID(e->msi.address_hi);
 	irq->vector = (e->msi.data &
 			MSI_DATA_VECTOR_MASK) >> MSI_DATA_VECTOR_SHIFT;
-	irq->dest_mode = (1 << MSI_ADDR_DEST_MODE_SHIFT) & e->msi.address_lo;
+	irq->dest_mode = (1 << MSI_ADDR_DEST_MODE_SHIFT) & e->msi.address_lo ?
+	    APIC_DEST_LOGICAL : APIC_DEST_PHYSICAL;
 	irq->trig_mode = (1 << MSI_DATA_TRIGGER_SHIFT) & e->msi.data;
 	irq->delivery_mode = e->msi.data & 0x700;
 	irq->msi_redir_hint = ((e->msi.address_lo
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 3ed167e039e5..3b00d662dc14 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -7356,7 +7356,7 @@ static void kvm_pv_kick_cpu_op(struct kvm *kvm, unsigned long flags, int apicid)
 	struct kvm_lapic_irq lapic_irq;
 
 	lapic_irq.shorthand = 0;
-	lapic_irq.dest_mode = 0;
+	lapic_irq.dest_mode = APIC_DEST_PHYSICAL;
 	lapic_irq.level = 0;
 	lapic_irq.dest_id = apicid;
 	lapic_irq.msi_redir_hint = false;

From patchwork Fri Nov 29 16:32:34 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Xu <peterx@redhat.com>
X-Patchwork-Id: 11267343
Return-Path: <SRS0=D48H=ZV=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 3AECE930
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 29 Nov 2019 16:32:49 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 145F0217AB
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 29 Nov 2019 16:32:49 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="jOc8isZJ"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727127AbfK2Qcr (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 29 Nov 2019 11:32:47 -0500
Received: from us-smtp-delivery-1.mimecast.com ([205.139.110.120]:49898 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1727073AbfK2Qcq (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 29 Nov 2019 11:32:46 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1575045165;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=YdjqiuhcZ58rcqgBhxb9lvVnvVL8Yk36IVCo28ntw5s=;
        b=jOc8isZJQtlTkjAgo0VnrohVrs9mwWjowNCOj9dVO1CMdHU4r0o6IWHh+ucU5z51auH+cu
        3w5wIf3QyMxXtXq1ZH8QrCc93A+XrYV+thZq+mrYD3U0O6XwEeZ4gAb3uNs4Aj2s8QZWEW
        Q1infbgLt83eopcDuXv1WJRAL/9y+U0=
Received: from mail-qv1-f72.google.com (mail-qv1-f72.google.com
 [209.85.219.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-432-d-euL8O8OjmJU3SUxenDWg-1; Fri, 29 Nov 2019 11:32:42 -0500
Received: by mail-qv1-f72.google.com with SMTP id bt18so2307963qvb.19
        for <kvm@vger.kernel.org>; Fri, 29 Nov 2019 08:32:42 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=pF321W2AO34u1L6FQdPVoWqPwEEaGFwdyk4B7rVy65s=;
        b=ixXFMXfiors/Ee1Rvd2Q+MO6xdZVTx1ZGGxmsHaW0EnZB3baAKrjObpTHnFYyIT3LD
         iB60LS2g3MTx806ns4wweeHp9wdQccmRNXwYVAWr6LrQ8LFiNaOPj8YL6VMblNgs96ZA
         7iAPJNm6fhSCvB4al/tAP3Gd08hdxK0wzMyhNw4DH8c9i88cYb4nVOmAYzeIyMQdirme
         0hqIRifVZLSCNjAe9xtVl+scLwEBm7xfhZwbxhpCwcA3Vt9VPgDF9D3NZoZVc0wh5u0Y
         ++R56Hu9RTymrome7A98twOIqKfJdlSAyb/lXL8aKX+C1ni6AKap7sZ0loCZYDO4L6hJ
         k8zQ==
X-Gm-Message-State: APjAAAVk0jmsoM2Y4wwCU39ly4HpBJzKyVgZCLLCUQ/a0QGwqLoug8uH
        3+tZBgxAWp48RrUY0QOgyGIxhj7jnMnJ6mT22QkKAycc4tzTvY/lkBqlXlZZ3lLEa5DxXS+ak90
        COP+yZ0jGxtp7
X-Received: by 2002:a37:7dc2:: with SMTP id
 y185mr16812163qkc.380.1575045161648;
        Fri, 29 Nov 2019 08:32:41 -0800 (PST)
X-Google-Smtp-Source: 
 APXvYqxL2taDIIh634zxS4jLBOiDRC64GcK8w8AVHUAo6D3g58URlp/1EyloE/5HF/qs4zkbAkP1Jw==
X-Received: by 2002:a37:7dc2:: with SMTP id
 y185mr16812137qkc.380.1575045161402;
        Fri, 29 Nov 2019 08:32:41 -0800 (PST)
Received: from xz-x1.yyz.redhat.com ([104.156.64.74])
        by smtp.gmail.com with ESMTPSA id
 d9sm4568329qtj.52.2019.11.29.08.32.39
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 29 Nov 2019 08:32:39 -0800 (PST)
From: Peter Xu <peterx@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Cc: Nitesh Narayan Lal <nitesh@redhat.com>,
        Paolo Bonzini <pbonzini@redhat.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        peterx@redhat.com, Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [PATCH v2 3/3] KVM: X86: Fixup kvm_apic_match_dest() dest_mode
 parameter
Date: Fri, 29 Nov 2019 11:32:34 -0500
Message-Id: <20191129163234.18902-4-peterx@redhat.com>
X-Mailer: git-send-email 2.21.0
In-Reply-To: <20191129163234.18902-1-peterx@redhat.com>
References: <20191129163234.18902-1-peterx@redhat.com>
MIME-Version: 1.0
X-MC-Unique: d-euL8O8OjmJU3SUxenDWg-1
X-Mimecast-Spam-Score: 0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The problem is the same as the previous patch on that we've got too
many ways to define a dest_mode, but logically we should only pass in
APIC_DEST_* macros for this helper.

To make it easier, simply define dest_mode of kvm_apic_match_dest() to
be a boolean to make it right while we can avoid to touch the callers.

Signed-off-by: Peter Xu <peterx@redhat.com>
---
 arch/x86/kvm/lapic.c | 5 +++--
 arch/x86/kvm/lapic.h | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/x86/kvm/lapic.c b/arch/x86/kvm/lapic.c
index cf9177b4a07f..80732892c709 100644
--- a/arch/x86/kvm/lapic.c
+++ b/arch/x86/kvm/lapic.c
@@ -791,8 +791,9 @@ static u32 kvm_apic_mda(struct kvm_vcpu *vcpu, unsigned int dest_id,
 	return dest_id;
 }
 
+/* Set dest_mode to true for logical mode, false for physical mode */
 bool kvm_apic_match_dest(struct kvm_vcpu *vcpu, struct kvm_lapic *source,
-			   int short_hand, unsigned int dest, int dest_mode)
+			   int short_hand, unsigned int dest, bool dest_mode)
 {
 	struct kvm_lapic *target = vcpu->arch.apic;
 	u32 mda = kvm_apic_mda(vcpu, dest, source, target);
@@ -800,7 +801,7 @@ bool kvm_apic_match_dest(struct kvm_vcpu *vcpu, struct kvm_lapic *source,
 	ASSERT(target);
 	switch (short_hand) {
 	case APIC_DEST_NOSHORT:
-		if (dest_mode == APIC_DEST_PHYSICAL)
+		if (dest_mode == false)
 			return kvm_apic_match_physical_addr(target, mda);
 		else
 			return kvm_apic_match_logical_addr(target, mda);
diff --git a/arch/x86/kvm/lapic.h b/arch/x86/kvm/lapic.h
index 19b36196e2ff..c0b472ed87ad 100644
--- a/arch/x86/kvm/lapic.h
+++ b/arch/x86/kvm/lapic.h
@@ -82,7 +82,7 @@ int kvm_lapic_reg_write(struct kvm_lapic *apic, u32 reg, u32 val);
 int kvm_lapic_reg_read(struct kvm_lapic *apic, u32 offset, int len,
 		       void *data);
 bool kvm_apic_match_dest(struct kvm_vcpu *vcpu, struct kvm_lapic *source,
-			   int short_hand, unsigned int dest, int dest_mode);
+			   int short_hand, unsigned int dest, bool dest_mode);
 int kvm_apic_compare_prio(struct kvm_vcpu *vcpu1, struct kvm_vcpu *vcpu2);
 int kvm_irq_delivery_to_apic(struct kvm *kvm, struct kvm_lapic *src,
 			     struct kvm_lapic_irq *irq,
